### PR TITLE
Add support for xz bottles

### PR DESCRIPTION
**Pull request by kabel**
* Installs XZ during setup
* Finds the json file from xz bottles

### CODE REVIEW

**Comment by kabel** [comment]
Related to https://github.com/Homebrew/brew/pull/3641

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -583,6 +583,7 @@ def setup
       test "brew", "doctor"
       test "brew", "--env"
       test "brew", "config"
+      test "brew", "install", "xz"
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: This should be conditional on if it's already installed, if `!ARGV.include?("--no-bottle")` and just before the first bottle is created https://github.com/kabel/homebrew-test-bot/blob/a8e82986e8ecbe70a02ddf540aaefbf7720afff9/cmd/brew-test-bot.rb#L628

**Comment by kabel** [comment]
Any new feedback or help on how to fix linux not being able to install xz?

**Comment by MikeMcQuaid** [comment]
Add an `if OS.mac?` or make it conditional on `Formula["xz"]` returning a value rather than throwing an exception.

**Comment by MikeMcQuaid** [comment]
Passing on this due to reasons explained in https://github.com/Homebrew/brew/pull/3641#issuecomment-363020837. Please take any conversation there.

**Pull request closed without merging**